thingif/src/main/java/com/kii/thingif/internal/gson/TimeRangeAdapter.java

```diff
@@ -0,0 +1,34 @@
+package com.kii.thingif.internal.gson;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.kii.thingif.query.TimeRange;
+
+import java.lang.reflect.Type;
+import java.util.Date;
+
+public class TimeRangeAdapter implements JsonDeserializer<TimeRange> {
+
+    @Override
+    public TimeRange deserialize(
+            JsonElement jsonElement,
+            Type typeOfT,
+            JsonDeserializationContext context)
+        throws JsonParseException
+    {
+        if (jsonElement == null) return null;
+
+        JsonObject json = jsonElement.getAsJsonObject();
+
+        if (!json.has("from") || !json.has("to")) {
+            throw new JsonParseException("no neceessary fields.");

```

Comment:
`no neceessary fields.` changed to `json doesn't contain both of 2 necessary fields: from and to. `